In [21]:
import sys
sys.path.append('../src')
from data.data_utils import load_partitions, read_partition

In [34]:
# GOOGLE COLLAB ONLY
from google.colab import drive

In [ ]:
drive.mount('/gdrive')

In [48]:
params = {
  'data_dir': '/gdrive/MyDrive/Netflix_Prize/processed',
  'output_dir': '/gdrive/MyDrive/Netflix_Prize/split',
  'test_ratio': 0.1,      # validation ratio
  'sample_ratio': 0.2     # sampling to preserve distribution
}


In [49]:
output = create_temporal_split(**params)

Found 34 partition files
Sampling dates from 6 files out of 34


Sampling dates: 100%|██████████| 6/6 [00:35<00:00,  5.93s/it]


Split date: 2005-10-03 00:00:00


Splitting files: 100%|██████████| 34/34 [00:46<00:00,  1.38s/it]


Completed temporal split


In [50]:
print(f'Split Date {output["split_date"]}')
for k, v in output['statistics'].items():
    print(f"{k}: {v}")

Split Date 2005-10-03 00:00:00
train_count: 88946246
validation_count: 10125866
total_count: 99072112
train_ratio: 0.8977929732637576
validation_ratio: 0.10220702673624238
train_files: 34
validation_files: 34
train_movies: 17770
validation_movies: 17770
movies_in_both: 17770
train_only_movies: 0
validation_only_movies: 0


We ended up with:
- ~89M training entries
- ~10.1M validation entries
- ~1.4M test (for final score)

Our splitting strategy looked to be successful. Notably all 17770 movies showed up in both the validation and the train datasets which is a sign that the distribution was kept relatively the same. As an extension we can look into different splitting strategies like validating on users/movies that weren't part of the training data.